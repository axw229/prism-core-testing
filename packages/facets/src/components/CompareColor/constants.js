export const baseClass = 'color__comparison'
export const containerClass = `${baseClass}__container`
export const containerHeaderClass = `${baseClass}__container__header`
export const containerHeaderButtonClass = `${baseClass}__container__header__button`
export const wrapperClass = `${baseClass}__wrapper`
export const prevBtnWrapperClass = `${baseClass}__prev-btn__wrapper`
export const buttonsClass = `${baseClass}__buttons`
export const buttonsVisibleClass = `${baseClass}__buttons--visible`
export const nextBtnWrapperClass = `${baseClass}__next-btn__wrapper`
export const backgroundColorClass = `${baseClass}__background__color`
export const queueWrapperClass = `${baseClass}__queue__wrapper`
