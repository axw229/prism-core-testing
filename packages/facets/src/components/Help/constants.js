export const baseClass = `cvw-help`
export const wrapper = `${baseClass}__wrapper`
export const tabsContainer = `${baseClass}__tabs-container`
export const contentWrapper = `${baseClass}__content-wrapper`
export const helpContent = `${baseClass}__help-content`
export const helpContentHide = `${helpContent}--hide`
export const contentHeader = `${baseClass}__content-header`
export const contentDetails = `${baseClass}__content-details`
export const subContent = `${baseClass}__subcontent`
export const subContentHeader = `${baseClass}__subContent-header`
export const subContentDetails = `${baseClass}__subContent-details`
export const helpIcons = `${baseClass}__help-icons`
export const helpImages = `${baseClass}__help-images`
export const helpImagesMobile = `${baseClass}__help-images-mobile`
export const iconWrap = `${baseClass}__icon-wrap`
export const iconWrapUndoRedo = `${baseClass}__icon-wrap-undoredo`
export const iconInfo = `${baseClass}__icon-info`
export const secondIcon = `${baseClass}__second-icon`
export const SCROLL_SPEED = 500
