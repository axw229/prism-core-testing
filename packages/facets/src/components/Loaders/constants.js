// @flow
// the 5 colors used by loaders in their default state; taken from SW CVW circa 2019
export const DEFAULT_COLORS = [
  '#00b9e4',
  '#69be28',
  '#ffb612',
  '#e17000',
  '#b71234'
]
// how many colors there are (this is sort of magic, so don't touch it)
export const NUM_COLORS = DEFAULT_COLORS.length
// how many generated colors are allowed to be dark
export const NUM_DARKS = Math.floor(NUM_COLORS / 2)
// how many generated colors should be light
export const NUM_LIGHTS = NUM_COLORS - NUM_DARKS
// default brightness step between generated colors
export const GENERATED_COLOR_BRIGHTNESS_STEP = 10
// % of brightness spectrum edges considered to be too dark or too light
export const GENERATED_COLOR_EDGE_PCT = 0.1
