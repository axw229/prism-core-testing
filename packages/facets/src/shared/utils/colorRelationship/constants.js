export const FULL = 360
export const HALF = FULL / 2
export const QUARTER = HALF / 2
export const NUM_CLUSTERS = 12
export const CLUSTER_SIZE = FULL / NUM_CLUSTERS
