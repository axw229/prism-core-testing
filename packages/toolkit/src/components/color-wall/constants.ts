export const MAX_SWATCH_SIZE = 50
export const BASE_SWATCH_SIZE = 15
export const OUTER_SPACING = 20
export const MIN_SCROLLER_HEIGHT = 200
export const MAX_SCROLLER_HEIGHT = MIN_SCROLLER_HEIGHT * 3

export const MIN_BASE_SIZE = 14
export const MAX_BASE_SIZE = 45
export const ZOOMED_BASE_SIZE = 50
export const ACTIVE_SWATCH_SIZE = 150
const BLOOMED_SWATCH_MIN_SIZE = 70
export const BLOOMED_SWATCH_SIZES = new Array(4).fill(null).map((v, i, arr) => {
  return BLOOMED_SWATCH_MIN_SIZE + ((ACTIVE_SWATCH_SIZE - BLOOMED_SWATCH_MIN_SIZE) / arr.length) * (arr.length - 1 - i)
})

export const SWATCH_WIDTH_WRAP_THRESHOLD = 20 // will prefer wrapped view if unwrapped view target swatch size falls below threshold value

export const TITLE_SIZE_RATIOS = {
  1: 0.4, // l1 should be styled regular weight
  2: 0.4, // l2 should be styled bold weight
  3: 0.5 // l3 should be styled bold weight, and is larger than l1 and l2
}

export const TITLE_SIZE_MIN = 12
export const TITLE_SIZE_MAX = 16
