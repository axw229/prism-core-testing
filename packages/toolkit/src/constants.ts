export enum SCENE_TYPES {
  OBJECT = 'objects',
  ROOM = 'rooms',
  AUTOMOTIVE = 'automotive',
  FAST_MASK = 'fastmask'
}

export enum DRAG_TYPES {
  SWATCH = 'SWATCH'
}

export enum SCENE_VARIANTS {
  DAY = 'day',
  NIGHT = 'night',
  MAIN = 'main'
}

// ERRORS
export const SYSTEM_ERROR = 'SYSTEM_ERROR'

export enum BUTTON_POSITIONS {
  TOP = 'top',
  BOTTOM = 'bottom'
}

export const KEY_CODES = {
  KEY_CODE_ARROW_DOWN: 40,
  KEY_CODE_ARROW_LEFT: 37,
  KEY_CODE_ARROW_RIGHT: 39,
  KEY_CODE_ARROW_UP: 38,
  KEY_CODE_ENTER: 13,
  KEY_CODE_SPACE: 32,
  KEY_CODE_TAB: 9
}
