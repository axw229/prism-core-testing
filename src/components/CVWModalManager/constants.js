export const GLOBAL_MODAL_STYLE = {
  success: 'success',
  danger: 'danger',
  warning: 'warning',
  primary: 'primary',
  secondary: 'secondary'
}

export const HANDLE_NAVIGATION_INTENT_CONFIRM = 'HANDLE_NAVIGATION_INTENT_CONFIRM'
export const HANDLE_NAVIGATION_INTENT_CANCEL = 'HANDLE_NAVIGATION_INTENT_CANCEL'
export const HANDLE_DIRTY_NAVIGATION_INTENT_CONFIRM = 'HANDLE_DIRTY_NAVIGATION_INTENT_CONFRIM'
export const HANDLE_DIRTY_NAVIGATION_INTENT_CANCEL = 'HANDLE_DIRTY_NAVIGATION_INTENT_CANCEL'
export const HANDLE_SELECT_PALETTE_CONFIRM = 'HANDLE_SELECT_PALETTE_CONFIRM'
export const HANDLE_DELETE_MY_PREVIEW_CONFIRM = 'HANDLE_DELETE_MY_PREVIEW_CONFIRM'
export const HIDE_MODAL = 'HIDE_MODAL'
export const DANGER = 'DANGER'
export const PRIMARY = 'PRIMARY'
export const SAVED = 'Saved'
export const ACTION_SAVE = 'SAVE'

// @todo may be redundant with MODAL_TYPE_ENUM -RS
export const SAVE_OPTION = {
  SAVE_STOCK_SCENE: 'SAVE_STOCK_SCENE_FROM_MODAL',
  SAVE_PAINT_SCENE: 'SAVE_PAINT_SCENE_FROM_MODAL',
  SAVE_LIVE_PALETTE: 'SAVE_LIVE_PALETTE_FROM_MODAL',
  EMPTY_SCENE: 'EMPTY_SCENE',
  SAVE_FAST_MASK: 'SAVE_FAST_MASK'
}

export const MODAL_TYPE_ENUM = {
  MATCH_PHOTO: 'MATCH_PHOTO',
  STOCK_SCENE: 'STOCK_SCENE',
  PAINT_SCENE: 'PAINT_SCENE',
  LIVE_PALETTE: 'LIVE_PALETTE',
  EMPTY_SCENE: 'EMPTY_SCENE',
  FAST_MASK: 'FAST_MASK'
}
