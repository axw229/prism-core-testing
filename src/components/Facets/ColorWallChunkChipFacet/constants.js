export const TYPES = {
  WALL: 'WALL',
  COL: 'COL',
  ROW: 'ROW',
  CHUNK: 'CHUNK',
  SECTION: 'SECTION',
  SWATCH: 'SWATCH'
}

export const BASE_SWATCH_SIZE = 30
export const CHUNK_SPACING = 14
