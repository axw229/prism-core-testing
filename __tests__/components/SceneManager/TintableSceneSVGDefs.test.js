import React from 'react'
import renderer from 'react-test-renderer'

import TintableSceneSVGDefs from '../../../src/components/SceneManager/TintableSceneSVGDefs'
import { SCENE_TYPES } from 'constants/globals'

const tintColors = [
  '#FF0000',
  '#0000FF'
]

describe('<TintableSceneSVGDefs />', () => {
  test('render empty without provided type', () => {
    const component = renderer.create(
      <TintableSceneSVGDefs />
    ).toJSON()

    expect(component).toEqual(null)
  })

  describe(`${SCENE_TYPES.ROOM} SVG filters and masks`, () => {
    test('render', () => {
      const component = renderer.create(
        <TintableSceneSVGDefs type={SCENE_TYPES.ROOM} />
      ).toJSON()

      expect(component).toMatchSnapshot()
    })

    test('does not respond to shadow and highlight maps', () => {
      const component1 = renderer.create(
        <TintableSceneSVGDefs type={SCENE_TYPES.ROOM} shadowMap={'shadows'} highlightMap={'highlights'}  />
      ).toJSON()

      const component2 = renderer.create(
        <TintableSceneSVGDefs type={SCENE_TYPES.ROOM} shadowMap={'other shadows'} highlightMap={'highlights'}  />
      ).toJSON()

      const component3 = renderer.create(
        <TintableSceneSVGDefs type={SCENE_TYPES.ROOM} shadowMap={'other shadows'} highlightMap={'other highlights'}  />
      ).toJSON()

      expect(component1).toEqual(component2)
      expect(component1).toEqual(component3)
      expect(component2).toEqual(component3)
    })

    test('feFlood is set to filterColor', () => {
      const component1 = renderer.create(
        <TintableSceneSVGDefs type={SCENE_TYPES.ROOM} filterColor={tintColors[0]}  />
      )

      const colorObject = component1.root.findAll((el) => {
        return el.type === 'feFlood' && el.props.floodColor === tintColors[0]
      })

      expect(colorObject).toHaveLength(1)
    })
  })

  describe(`${SCENE_TYPES.OBJECT} SVG filters and masks`, () => {
    test('render', () => {
      const component = renderer.create(
        <TintableSceneSVGDefs type={SCENE_TYPES.OBJECT}  />
      ).toJSON()

    })

    test('responds to shadow and highlight maps', () => {
      const component1 = renderer.create(
        <TintableSceneSVGDefs type={SCENE_TYPES.OBJECT} shadowMap={'shadows'} highlightMap={'highlights'}  />
      ).toJSON()

      const component2 = renderer.create(
        <TintableSceneSVGDefs type={SCENE_TYPES.OBJECT} shadowMap={'other shadows'} highlightMap={'highlights'}  />
      ).toJSON()

      const component3 = renderer.create(
        <TintableSceneSVGDefs type={SCENE_TYPES.OBJECT} shadowMap={'other shadows'} highlightMap={'other highlights'}  />
      ).toJSON()

      expect(component1).not.toEqual(component2)
      expect(component1).not.toEqual(component3)
      expect(component2).not.toEqual(component3)
    })

    test('feFlood is set to filterColor', () => {
      const component1 = renderer.create(
        <TintableSceneSVGDefs type={SCENE_TYPES.OBJECT} filterColor={tintColors[0]}  />
      )

      const colorObject = component1.root.findAll((el) => {
        return el.type === 'feFlood' && el.props.floodColor === tintColors[0]
      })

      expect(colorObject).toHaveLength(1)
    })
  })
})
