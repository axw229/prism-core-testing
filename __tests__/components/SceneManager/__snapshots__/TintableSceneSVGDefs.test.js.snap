// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TintableSceneSVGDefs /> objects SVG filters and masks render 1`] = `
<svg
  height="0"
  version="1.1"
  viewBox="0 0 undefined undefined"
  width="0"
  x="0"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
  y="0"
>
  <defs>
    <filter
      colorInterpolationFilters="sRGB"
      filterUnits="objectBoundingBox"
      height="100%"
      primitiveUnits="objectBoundingBox"
      width="100%"
      x="0"
      y="0"
    >
      <feFlood
        result="tintHue"
      />
      <feImage
        height="100%"
        result="highlight-map"
        width="100%"
        x="0"
        y="0"
      />
      <feImage
        height="100%"
        result="shadow-map"
        width="100%"
        x="0"
        y="0"
      />
      <feBlend
        in="tintHue"
        in2="highlight-map"
        mode="screen"
        result="tint-highlight-blend"
      />
      <feBlend
        in="shadow-map"
        in2="tint-highlight-blend"
        mode="multiply"
        result="finished-tint"
      />
      <feBlend
        in="finished-tint"
        in2="tintHue"
        mode="luminosity"
        result="finished-filter"
      />
    </filter>
    <mask
      height="100%"
      maskUnits="objectBoundingBox"
      width="100%"
      x="0"
      y="0"
    >
      <image
        height="100%"
        width="100%"
        x="0"
        y="0"
      />
    </mask>
  </defs>
</svg>
`;

exports[`<TintableSceneSVGDefs /> rooms SVG filters and masks render 1`] = `
<svg
  height="0"
  version="1.1"
  viewBox="0 0 undefined undefined"
  width="0"
  x="0"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
  y="0"
>
  <defs>
    <filter
      colorInterpolationFilters="sRGB"
      filterUnits="objectBoundingBox"
      height="100%"
      primitiveUnits="objectBoundingBox"
      width="100%"
      x="0"
      y="0"
    >
      <feComponentTransfer
        in="SourceGraphic"
        result="adjustedHistogram"
      >
        <feFuncR
          type="table"
        />
        <feFuncG
          type="table"
        />
        <feFuncB
          type="table"
        />
      </feComponentTransfer>
      <feFlood
        result="tintHue"
      />
      <feBlend
        in="adjustedHistogram"
        in2="tintHue"
        mode="multiply"
      />
    </filter>
    <mask
      height="100%"
      maskUnits="objectBoundingBox"
      width="100%"
      x="0"
      y="0"
    >
      <image
        height="100%"
        width="100%"
        x="0"
        y="0"
      />
    </mask>
  </defs>
</svg>
`;
